### ISSUE TITLE
Add additional constructor(s) CompositeStateRepository so it works better with lists

### ISSUE THREAD

**Issue by marceloverdijk** [togglz/togglz#147]
The currecnt `CompositeStateRepository` constructor does not work well if you have a list (or even arrray) of state repositories.

The only constructor is has is now:

```
public CompositeStateRepository(StateRepository repository,
            StateRepository... repositories) {

        this.repositories = merge(repository, repositories);
    }
```

Maybe good to add something like:

```
public CompositeStateRepository(StateRepository... repositories) {

        this.repositories = merge(repository, repositories);
    }
```

**Comment by chkal** [comment]
Unfortunately I don't think this is possible as both constructors are overlapping. Which of the two should the JVM execute when you call:

```java
new CompositeStateRepository(a,b,c);
```

But I agree that the constructor is very nice. Perhaps we could replace the old one with the new one suggested by you without breaking compatibility?

**Comment by marceloverdijk** [comment]
Yes fair enough, as I think the current constructor is not so nice practical.

**Comment by chkal** [comment]
Fixed by #150
